static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.tsx

```diff
@@ -48,6 +51,29 @@ function WidgetBuilderV2({
   const {selection} = usePageFilters();
 
   const [queryConditionsValid, setQueryConditionsValid] = useState<boolean>(true);
+  const theme = useTheme();
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+  const [isPreviewDraggable, setIsPreviewDraggable] = useState(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowWidth(window.innerWidth);
+    };
+
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  const isSmallScreen =
```

Comment:
It looks like there's a hook for this called `useMedia`, if you check the code base for its usage we could probably use that?

Reply:
I think that also means we can get rid of the `useEffect` here, right? Let's see if that could be removed in the other file too

Reply:
oh no wayyy, that's super helpful